File: javascript/2015/day08b/day08b.js
#!/usr/bin/env node

const fs = require('fs');
const path = require('path');
const process = require('process');

function usage() {
    const progname = path.basename(process.argv[1]);
    console.log(`usage: ${progname} <input file>`);
    process.exit(1);
}

function processFile(filename) {
    let result = 0;
    const data = fs.readFileSync(filename, 'utf8');
    const lines = data.trim().split('\n');
    for ( const line of lines ) {
        const codeLen = line.length;
        var encLen = 0;
        for ( let i = 0; i < line.length; i++ ) {
            switch ( line[i] ) {
                case '\\':
                case '"':
                    encLen += 2;
                    break;
                default:
                    encLen++;
            }
        }
        result += 2 + (encLen - codeLen);
    }
    return result;
}

function main() {
    if ( process.argv.length < 3 ) {
        usage();
    }
    const filename = process.argv[2];
    const result = processFile(filename);
    console.log(`result = ${result}`);
}

main()

File: ruby/2015/day08b/day08b.rb
#!/usr/bin/env ruby

def usage
    progname = File.basename($0)
    puts "usage: #{progname} <input file>"
    exit 1
end

def process(filename)
    result = 0
    File.foreach(filename) do |line|
        code_len = line.length
        enc_len = 0
        for i in 0..(line.length - 1) do
            case line[i]
            when "\\", "\""
                enc_len += 2
            else
                enc_len += 1
            end
        end
        result += 2 + (enc_len - code_len)
    end
    result
end

def main
    usage if ARGV.length < 1
    filename = ARGV[0]
    result = process(filename)
    puts "result = #{result}"
end

main if __FILE__ == $0
